src/server/handlers/onReferences.ts

```diff
@@ -0,0 +1,151 @@
+import { getContext } from '../context';
```

Comment:
Perhaps extraction `onReferences` handler into separate file would be better to do in separate commit since it doesn't look relevant  here.